interview-questions.md

```diff
@@ -6,92 +6,121 @@ First, without external resources. Challenge yourself to answer from memory as i
+- `this` is a syntax used to call an object in this.state or from this.props
 
+Researched answer:
 
+- `this` is determined by how a function is called, refrences to an object and sometimes can be a value, can be global, a function.
 
 9. STRETCH: What is a ternary operator in JavaScript?
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+- A simplified syntax of an if/ else statement.
 
+Researched answer:
 
+- Ternary operator should evaluate boolean value, ? will seperate truthy outcome, and a : will seperate truthy from the falsey outcome. Used for refactoring.
 
 10. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+- anonymous function is a a method used without having to specifically call it.
 
+Researched answer:
 
+- Anonymous function is a function without a name, but not always accessable after initial creation
 
 ## Looking Ahead: Terms for Next Week
-- Ruby
-- Object oriented programming
-- RSpec
-- Instance variable
-- Ruby blocks
-- Ruby hashes
-- `getter` and `setter` methods in Ruby
+
+- Ruby: Is an object oriented language, preinstalled into macs, Ruby is a general-purpose, interpreted, dynamically typed, server-side scripting language, can be embedded in HTML, uses Object Oriented Programming.
+
+- Object oriented programming: OOP was created to deal with complex software to deal with how fast applications were growing in complexity. You are able to maintain a better app when working on smaller component/ objects it doesnt effect the main parent code.
+
+- RSpec:Uses red-green-refactoring for test driven development, its composed of multiple libraries, its flexible and customizable.
+
+- Instance variable: Is a type of class atribute.
+- Ruby blocks: are anonymous functions in Ruby, created with do/end keywords or {}. Its a grouping of statements that are written on the same line as the method calls last parameter.
+
+- Ruby hashes: Is a collection of keys and their value pairs, similar to and array and also can use iterable methods on a hash.
+- `getter` and `setter` methods in Ruby: They allow to get the classes instance variable outside the class,`getter` methods are used to `get` the value of an instance variable and binds an object property to a function that can be called upon. While `setter` methods are used to `set` the value of an instance variable and binds an object property to a function to be called when there is an attempt to set the property.
+
+P.S. I utilized Sarah, Brian and my fellow classmates a lot through this, would not have been able to finish the code-challenges.js without their help. I am focusing heavily this week on going through the syllabus and create flash cards for myself to start instilling the fundamentals deeper in my brain. As of now I can see the logic and even talk about it, but when I try and type it out, its a blank wall. very difficult for me. but its all about the push for greatness, and this week will become better!
```

Comment:
totally! use the resources you have available to you. but im glad to hear that you'er putting in the extra work to ensure your success.